core/src/components/modal/modal.tsx

```diff
@@ -705,17 +705,26 @@ export class Modal implements ComponentInterface, OverlayInterface {
     }
 
     const enteringAnimation = activeAnimations.get(this) || [];
-
-    this.currentTransition = dismiss(this, data, role, 'modalLeave', iosLeaveAnimation, mdLeaveAnimation, {
-      presentingEl: this.presentingElement,
-      currentBreakpoint: this.currentBreakpoint !== undefined || this.initialBreakpoint,
-      backdropBreakpoint: this.backdropBreakpoint,
-    });
+    const { delegate, inline } = this.getDelegate();
+
+    this.currentTransition = dismiss(
+      this,
+      data,
+      role,
+      'modalLeave',
+      iosLeaveAnimation,
+      mdLeaveAnimation,
+      inline === false || delegate === undefined,
```

Comment:
Can we extract this to a variable that better defines what it is checking? My understanding is we want to only remove the overlay if the overlay is presented via a controller, right?

Reply:
Correct, the overlay should only be removed if presented via a controller.

Should it be something like:
```ts
const hasController = inline === false || delegate === undefined;
```

or explicit to the removal of the overlay element?

```ts
const removeOverlay = inline === false || delegate === undefined;
```